.ci/Jenkinsfile

```diff
@@ -0,0 +1,187 @@
+    PATH = "${env.PATH}:${env.WORKSPACE}/bin"
+    PIPELINE_LOG_LEVEL = 'INFO'
+    ECS_REPO = "${params?.ECS_REPO}"
+  }
+  options {
+    timeout(time: 3, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '5', artifactNumToKeepStr: '5'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+  }
+  triggers {
+    cron('H H(4-5) * * 1,5')
+  }
+  parameters {
+    booleanParam(name: 'DRY_RUN_MODE', defaultValue: false, description: 'If true, allows to execute this pipeline in dry run mode, without sending a PR.')
+    booleanParam(name: 'FORCE_SEND_PR', defaultValue: false, description: 'If true, will force sending a PR, although it could be affected by the value off the DRY_RUN parameter: if the latter is true, a message will be printed in the console.')
+    choice(name: 'ECS_REPO', choices: ['all',
+                                        'ecs-dotnet',
+                                        'ecs-logging-go-logrus',
+                                        'ecs-logging-go-zap',
+                                        'ecs-logging-java',
+                                        'ecs-logging-nodejs',
+                                        'ecs-logging-python',
+                                        'ecs-logging-ruby'], description: 'Name of the ECS repo you want to update its specs.')
+  }
+  stages {
+    stage('Checkout'){
+      steps {
+        deleteDir()
+        gitCheckout(basedir: "${BASE_DIR}",
+          repo: "git@github.com:elastic/${REPO}.git",
+          credentialsId: "${JOB_GIT_CREDENTIALS}"
+        )
+        stash allowEmpty: true, name: 'source', useDefaultExcludes: false
+      }
+    }
+    // This stage will populate the environment, and will only be executed under any of the
+    // following conditions:
+    // 1. we run the pipeline NOT in DRY_RUN_MODE, because we want to send real PRs
+    // 2. we run the pipeline forcing sending real PRs, because we want so
+    // Because the rest of the following stages will need these variables to check for changes,
+    // skipping this stage would not take effect in them, as they are covered by the 
+    // FORCE_SEND_PR check.
+    stage('Check for spec changes'){
+      when {
+        beforeAgent true
+        anyOf {
+          expression { return env.DRY_RUN_MODE == "false" }
+          expression { return params.FORCE_SEND_PR }
+        }
+      }
+      environment {
+        // GIT_PREVIOUS_SUCCESSFUL_COMMIT might point to a local merge commit instead a commit in the
+        // origin, then let's use the target branch for PRs and the GIT_PREVIOUS_SUCCESSFUL_COMMIT for
+        // branches.
+        COMMIT_FROM = """${isPR() ? "origin/${env.CHANGE_TARGET}" : "${env.GIT_PREVIOUS_SUCCESSFUL_COMMIT}"}"""
+      }
+      steps {
+        deleteDir()
+        unstash 'source'
+        script {
+          dir("${BASE_DIR}"){
+            regexps =[
+              "^spec/.*"
```

Comment:
I'm not sure I follow this code exactly, but will this result in SPECS_UPDATED being true if there is only a change to *README.md* in https://github.com/elastic/ecs-logging/tree/master/spec ?

Reply:
Indeed, though it's harmless since the one in charge to copy files is the `sh` script, that only cares for spec.json files. This is regarding the filtering of any pushes in the project, and discard those commits that did not change the `spec` folder.

Even though if a README.md file got changed the PRs won't be created.

I'll change it for clarity, thanks for the heads up

Reply:
```suggestion
              "^spec/spec.json"
```